registry/session_v2.go

```diff
@@ -124,14 +124,15 @@ func (r *Session) HeadV2ImageBlob(ep *Endpoint, imageName, sumType, sum string,
 		return false, err
 	}
 	res.Body.Close() // close early, since we're not needing a body on this call .. yet?
-	switch res.StatusCode {
-	case 200:
+	switch {
+	case res.StatusCode >= 200 && res.StatusCode < 400:
```

Comment:
I find it a bit counterintuitive that things outside 200 is interpreted as success. Can you explain a bit more? How for example would a 301 be understood as "no push needed"?